test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests/PackagesConfigProjectTests/PackagesLockTests.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Configuration;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+using NuGet.Versioning;
+using Test.Utility;
+using Xunit;
+
+namespace NuGet.PackageManagement.Test.NuGetPackageManagerTests.PackagesConfigProjectTests
+{
+    public class PackagesLockTests
+    {
+        private readonly List<PackageIdentity> PackageWithDependents = new List<PackageIdentity>
+            {
+                new PackageIdentity("jQuery", new NuGetVersion("1.4.4")),
+                new PackageIdentity("jQuery", new NuGetVersion("1.6.4")),
+                new PackageIdentity("jQuery.Validation", new NuGetVersion("1.13.1")),
+                new PackageIdentity("jQuery.UI.Combined", new NuGetVersion("1.11.2"))
+            };
+
+
+        [Fact]
+        public async Task DoNotCreateLockFileWhenFeatureDisabled()
+        {
+            // Arrange
+            var sourceRepositoryProvider = TestSourceRepositoryUtility.CreateV3OnlySourceRepositoryProvider();
```

Comment:
don't use http source instead create local source to run your tests. 

I know some of the existing tests are still pointing to V3 sources but all new tests are being added with their own local source and clean afterwards.